src/Grids/convenience_constructors.jl

```diff
@@ -0,0 +1,366 @@
+#=
+While we recommend users to use our composable
+methods, this file contains some convenience
+constructors, which are meant to improve
+developer experience.
+=#
```

Comment:
I think we should recommend using the compasable constructors only if one needs them, and try using these constructors when possible

Reply:
I've updated the doc string for this, I think it's an improvement. Can you take another look?

Reply:
Update looks good